src/main/java/appeng/helpers/DualityInterface.java

```diff
@@ -1275,10 +1276,34 @@ else if( capabilityClass == Capabilities.STORAGE_MONITORABLE_ACCESSOR )
 
 	private class InterfaceRequestSource extends MachineSource
 	{
+		private final InterfaceRequestContext context;
 
-		public InterfaceRequestSource( final IActionHost v )
+		public InterfaceRequestSource( IActionHost v )
 		{
 			super( v );
+			this.context = new InterfaceRequestContext();
+		}
+
+		@Override
+		public <T> Optional<T> context( Class<T> key )
+		{
+			if( key == InterfaceRequestContext.class )
+			{
+				return (Optional<T>) Optional.of( this.context );
+			}
+
+			return super.context( key );
+		}
+
+	}
+
+	private class InterfaceRequestContext implements Comparable<Integer>
+	{
+
+		@Override
+		public int compareTo( Integer o )
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override "equals(Object obj)" to comply with the contract of the "compareTo(T o)" method. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1210)